### DISCUSSION TITLE
About Skywalking-go display problem of Endpoint and URL on trace

### DISCUSSION THREAD

**Discussion by CodePrometheus** [apache/skywalking#11807] [General]
Hi Skywalking community, long time no see!

I found that for many web plugins in skywalking-go, when I request an unintercepted route, the endpoint is not displayed, as shown in Figure 1, but in skywalking-java, such as spring-web, it's displayed as shown in Figure 2. I think it is possible to align skywalking-java. I'm willing to submit a PR to fix it.

<img width="302" alt="image1" src="https://github.com/apache/skywalking/assets/66550292/3ccd6011-79e4-4118-bcf4-63268e7f03e8">

<img width="319" alt="image2" src="https://github.com/apache/skywalking/assets/66550292/a0356cc3-8b95-4e3f-9399-40bb738de3eb">

**Comment by wu-sheng** [comment]
I hink that depends on the plugin implementation, rather than general skywalking go agent.

**Reply by CodePrometheus** [reply]
Yes, the method in some plugins is to return an empty string when requesting an undefined route. I think it would be better to display it, and I would like to know what the community members think.

**Reply by wu-sheng** [reply]
So, are you planning to put URI or something as endpoint name?

**Reply by CodePrometheus** [reply]
Just adjust the value of operationName in the plugin. When requesting a route with no defined interception, do not use an empty string. For example, in gin, FullPath() is currently used, and I want to adjust it to Request.URL.Path.
